### PR TITLE
Added delete user endpoint

### LINKED ISSUE #317: Add delete user
## Feature Title

Add a delete user use case.

## Acceptance Criteria

- [ ] Use case
- [ ] Router
- [ ] Remove user from the cognito

## Technical Details

- API Changes:
- Database Changes:
- Dependencies:

## Additional Context

None

### PR DESCRIPTION
**Pull request by hitohata**
## Overviews of implementation

Added a delete user endpoint

## Review points

Check the test if this is enough

### CODE REVIEW

File: backend/main/src/Services/DeleteUserService/DeleteUserService.ts [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+interface IDeleteUserProps {
+	/** requested by the user */
+	userId: string;
+	/** from the token */
+	targetUserId: string;
+}
+
+/**
+ * Delete a user.
+ * The groups and containers that are possessed by only this user.
```
Comment (taniguchiiqqq) [inline backend/main/src/Services/DeleteUserService/DeleteUserService.ts]: To make it clear, it seems better to be explicit that the groups and containers are deleted.
```suggestion
 * The groups and containers are deleted when they are possessed by only this user.
```

File: backend/main/src/Services/DeleteUserService/DeleteUserService.ts [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+	ListUsersCommandInput,
+} from "@aws-sdk/client-cognito-identity-provider";
+import { COGNITO_USER_POOL_ID } from "src/Settings/Setting";
+import { Err, Ok, Result } from "result-ts-type";
+
+interface IDeleteUserProps {
+	/** requested by the user */
+	userId: string;
+	/** from the token */
+	targetUserId: string;
```
Comment (taniguchiiqqq) [inline backend/main/src/Services/DeleteUserService/DeleteUserService.ts]: I think the roles of each prop are opposite. targetUserId is the target user of deletion specified by path parameter.

File: backend/main/src/Services/DeleteUserService/DeleteUserService.ts [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+	constructor(
+		readonly groupRepository: IGroupRepository,
+		readonly userRepository: IUserRepository,
+		readonly groupQuery: FindGroupsInformationQuery,
+	) {
+		super();
+	}
+
+	async handler(input: IDeleteUserProps) {
+		// check the requested user token and the user token's one
```
Comment (taniguchiiqqq) [inline backend/main/src/Services/DeleteUserService/DeleteUserService.ts]: I feel it's better to use the word user ID cuz the comparison is not about tokens but IDs.
```suggestion
		// check the requested user ID is the same as the user ID from the auth token.
```

File: backend/main/src/Services/DeleteUserService/DeleteUserService.ts [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+
+		// check user ID
+		const userIdOrError = UserId.create(input.userId);
+		if (userIdOrError.err) {
+			return this.badRequest(userIdOrError.error);
+		}
+
+		const userId = userIdOrError.value;
+
+		// check if the user is the owner of the group
```
Comment (taniguchiiqqq) [inline backend/main/src/Services/DeleteUserService/DeleteUserService.ts]: Could you change this comment? cuz this check is in the `removeUserFromNishiki` function. So maybe this comment may suppose to be `check if the user exist`

File: backend/main/src/Services/DeleteUserService/DeleteUserService.ts [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+	}
+
+	async handler(input: IDeleteUserProps) {
+		// check the requested user token and the user token's one
+		if (input.userId !== input.targetUserId) {
+			return this.forbidden("You cannot delete another user");
+		}
+
+		// check user ID
+		const userIdOrError = UserId.create(input.userId);
```
Comment (taniguchiiqqq) [inline backend/main/src/Services/DeleteUserService/DeleteUserService.ts]: I think it should be targetUserId.

**Review by taniguchiiqqq — APPROVED** [review]
Thank you so much for implementing this feature. And so sorry for the late review. I think the test cases for the abnormal systems is enough.

**Review by taniguchiiqqq — CHANGES_REQUESTED** [review]
Sorry I mistakenly approved, please check the comments

**Comment by hitohata** [comment]
@taniguchiiqqq 
Thank you for your comment. I've updated. Could you please check it when you have a free time?

**Review by taniguchiiqqq — APPROVED** [review]
Thanks!! LGTM

**Pull request merged**